jraft-core/src/main/java/com/alipay/sofa/jraft/core/Replicator.java

```diff
@@ -200,6 +199,86 @@ private ReplicatorMetricSet(final ReplicatorOptions opts, final Replicator r) {
         INSTALLING_SNAPSHOT // installing snapshot
     }
 
+    enum ReplicatorEvent {
+        CREATED, // created
+        ERROR, // error
+        DESTROYED // destroyed
+    }
+
+    /**
+     * User can implement the ReplicatorStateListener interface by themselives,
+     * So they can do some their own logic codes when replicator started ,stopped or has some errors.
+     *
+     * @author zongtanghu
+     *
+     * 2019-Aug-20 2:32:10 PM
+     */
+    public interface ReplicatorStateListener {
+
+        /**
+         * Called when this replicator is started.
+         *
+         * @param peer   replicator related peerId
+         */
+        void onCreated(final PeerId peer);
+
+        /**
+         * Called when this replicator has some errors
+         *
+         * @param peer   replicator related peerId
+         * @param status replicator's error detailed status
+         */
+        void onError(final PeerId peer, final Status status);
+
+        /**
+         * Called when this replicator has stopped
+         *
+         * @param peer   replicator related peerId
+         */
+        void onDestroyed(final PeerId peer);
+    }
+
+    /**
+     * Notify replicator event(such as created, error, destroyed) to replicatorStateListener which is implemented by users
+     *
+     * @param replicator    replicator object
+     * @param event         replicator's state listener event type
+     * @param status        replicator's error detailed status
+     */
+    private static void notifyReplicatorStatusListener(final Replicator replicator, final ReplicatorEvent event, final Status status) {
+        Requires.requireNonNull(replicator.getOpts(), "ReplicatorOptions");
+        Requires.requireNonNull(replicator.getOpts().getNode(), "Node");
+        Requires.requireNonNull(replicator.getOpts().getPeerId(), "Peer");
+
+        PeerId peer = replicator.getOpts().getPeerId();
+        List<ReplicatorStateListener> listenerList = replicator.getOpts().getNode().getReplicatorStatueListeners();
+        for (int i = 0; i< listenerList.size(); i++) {
+            final ReplicatorStateListener listener = listenerList.get(i);
+            if(listener != null) {
+                try {
+                    switch (event) {
+                        case CREATED: {
+                            Utils.runInThread(()->listener.onCreated(peer));
+                            break;
+                        }
+                        case ERROR: {
+                            Utils.runInThread(()->listener.onError(peer, status));
+                            break;
+                        }
+                        case DESTROYED: {
+                            Utils.runInThread(()->listener.onDestroyed(peer));
+                            break;
+                        }
+                        default:
+                            break;
+                    }
+                } catch (Exception e) {
```

Comment:
final Exception e

Reply:
Okay,I will fix this codes comments and then push it again.Thanks!